.github/actions/build-android/action.yml

```diff
@@ -0,0 +1,104 @@
+name: Android Build
+description: Installs required dependencies and then builds the Android side of the SDK 
+
+inputs:
+  node-version:
+    description: The version of Node.js to use
+    required: true
+  java-version:
+    description: The version of Java to use
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Dependencies including 32-bit support
+      run: |
+        sudo dpkg --add-architecture i386
+        sudo apt-get update -y -qq
+        sudo apt-get install -y -qq gperf libatomic1:i386 libc6:i386 libncurses5:i386 libstdc++6:i386
+      shell: bash
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: 'npm'
+
+    - name: Setup Java
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: ${{ inputs.java-version }}
+
+    - name: Install dependencies
+      run: npm ci
+      shell: bash
```

Comment:
Because these files are for [composite actions](https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/) which are intended to be shared and reused (as we do here between the build and release workflows), you have linked to a workflow action file which is not the same thing. 

As the ADR I linked to states "The shell attribute is required for each run step because the action author does not know what the workflow author is using for the operating system so we need to explicitly prevent unknown behavior by making sure that each run step has an explicit shell set by the action author"